src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -9,6 +9,20 @@ public class RegistrationServiceImpl implements RegistrationService {
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new UserDataInvalidException("The user provided is null");
+        }
+
+        if (checkDataValidity(user)) {
+            return storageDao.add(user);
+        }
+        throw new UserDataInvalidException("The user data is not valid");
+    }
+
+    private boolean checkDataValidity(User user) {
+        return user.getLogin().length() >= 6
+                && user.getPassword().length() >= 6
+                && user.getAge() >= 18
+                && storageDao.get(user.getLogin()) == null;
```

Comment:
Split the single check in `checkDataValidity` into multiple if statements. This will make it easier to identify which specific data is invalid and to return a more informative error message.